src/crypto/cn_heavy_hash_hard_arm.cpp

```diff
@@ -29,14 +29,29 @@
 // Parts of this file are originally copyright (c) 2014-2017, The Monero Project
 // Parts of this file are originally copyright (c) 2012-2013, The Cryptonote developers
 
+#if defined(__aarch64__)
+
+#ifndef __clang__
+#  pragma GCC target ("+crypto")
+#endif
+
 #include "cn_heavy_hash.hpp"
 extern "C" {
 #include "keccak.h"
 #include "jh.h"
 #include "skein.h"
 }
 
-#ifdef HAS_ARM_HW
+#include <sys/auxv.h>
+#include <asm/hwcap.h>
+#include <arm_neon.h>
+
+static bool hw_check_aes()
+{
+	return (getauxval(AT_HWCAP) & HWCAP_AES) != 0;
+}
+
+extern "C" const bool cpu_aes_enabled = hw_check_aes() && check_override();
```

Comment:
I think that `&& check_override()` is meant to be `!check_override()`

Reply:
Oops! Fixed it up.  Also the strcmp's in it always confuse me, but I'm pretty sure they are right: return true if the env variable is set and i'ts not equal to 0 and not equal to no.

Reply:
Same got confused had to sit down and write out a truth table :)